### ISSUE TITLE
Allow to configure location of boilerplate and input/output files

### ISSUE THREAD

**Issue by king-of-poppk** [MangoTheCat/callr#262]
As a follow-up question to #261, I'd like to be able to use [`bubblewrap`](https://github.com/containers/bubblewrap) to isolate the different `callr` subprocesses coming from the same parent process. Currently it seems all `callr::r` calls will reference the same `.so` files and read inputs and write outputs to the same temporary folder.

Would it be possible to store the reused files in a separate folder (so that that one can be mounted read-only), and the input/output files in another folder so that that one can be mounted read-write separately?

Additionally would it be possible to have an easy way to know and/or configure where those are located?

There are probably cross-platform concerns there too maybe, although I am not sure whether many folders vs one is so big of a deal.

**Comment by gaborcsardi** [comment]
Yes, this is a duplicate of #172.

**Comment by king-of-poppk** [comment]
@gaborcsardi Since #172 was closed with #316 which does not really answer the issue described here, can I reopen?

Shortcomings of #316 wrt the present issue: `CALLR_TMPDIR` is configurable, yes, but configuration needs to happen before loading `callr`, and cannot be changed after loading `callr` without breaking things. This prevents using a different directory for each call (the `callr` subprocess needs to have write access).
